From 3de8752a4e9d346f6742fe7b28dbae34fe14925e Mon Sep 17 00:00:00 2001
From: Xu Xuehui <xxh@rock-chips.com>
Date: Wed, 16 Aug 2017 11:36:47 +0800
Subject: [PATCH] net: wireless: rockchip_wlan: add network wakeup function for
 rtl8723bs

wifi driver open config such as below:

CONFIG_WOWLAN = y
CONFIG_GPIO_WAKEUP = y

Change-Id: I9bdb07e0210c68993bb638aa2ca403142e180c5f
Signed-off-by: Xu Xuehui <xxh@rock-chips.com>
---
 .../rtl8723bs/platform/platform_ops.c         | 77 ++++++++++---------
 1 file changed, 40 insertions(+), 37 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rtl8723bs/platform/platform_ops.c b/drivers/net/wireless/rockchip_wlan/rtl8723bs/platform/platform_ops.c
index e9b0191f8bb8..4ea25ba852c3 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8723bs/platform/platform_ops.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8723bs/platform/platform_ops.c
@@ -1,37 +1,40 @@
-/******************************************************************************
- *
- * Copyright(c) 2013 Realtek Corporation. All rights reserved.
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of version 2 of the GNU General Public License as
- * published by the Free Software Foundation.
- *
- * This program is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
- * more details.
- *
- * You should have received a copy of the GNU General Public License along with
- * this program; if not, write to the Free Software Foundation, Inc.,
- * 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA
- *
- *
- ******************************************************************************/
-#ifndef CONFIG_PLATFORM_OPS
-/*
- * Return:
- *	0:	power on successfully
- *	others: power on failed
- */
-int platform_wifi_power_on(void)
-{
-	int ret = 0;
-
-
-	return ret;
-}
-
-void platform_wifi_power_off(void)
-{
-}
-#endif // !CONFIG_PLATFORM_OPS
+/******************************************************************************
+ *
+ * Copyright(c) 2013 Realtek Corporation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA
+ *
+ *
+ ******************************************************************************/
+#ifndef CONFIG_PLATFORM_OPS
+
+/* Return:
+ *	0:	power on successfully
+ *	others: power on failed
+ */
+#include <linux/rfkill-wlan.h>
+extern unsigned int oob_irq;
+int platform_wifi_power_on(void)
+{
+	int ret = 0;
+
+	oob_irq = rockchip_wifi_get_oob_irq();
+
+	return ret;
+}
+
+void platform_wifi_power_off(void)
+{
+}
+#endif // !CONFIG_PLATFORM_OPS
-- 
2.35.3

